qa/integration-tests/src/test/java/io/camunda/it/migration/util/MigrationITExtension.java

```diff
@@ -0,0 +1,306 @@
+            new ParameterResolver() {
+              @Override
+              public boolean supportsParameter(
+                  final ParameterContext parameterContext, final ExtensionContext extensionContext)
+                  throws ParameterResolutionException {
+                return Set.of(
+                        DatabaseType.class,
+                        CamundaClient.class,
+                        CamundaMigrator.class,
+                        DocumentBasedSearchClient.class,
+                        ExtensionContext.class)
+                    .contains(parameterContext.getParameter().getType());
+              }
+
+              @Override
+              public Object resolveParameter(
+                  final ParameterContext parameterContext, final ExtensionContext extensionContext)
+                  throws ParameterResolutionException {
+                final Class<?> requestedClass = parameterContext.getParameter().getType();
+                final var paramSupplier = resolveParam(requestedClass, extensionContext);
+                if (paramSupplier != null) {
+                  return paramSupplier;
+                }
+                throw new ParameterResolutionException("Unsupported parameter type");
+              }
+            });
+      }
+    };
+  }
+
+  private void startDatabaseContainer(final DatabaseType db, final Network network) {
+    switch (db) {
+      case ELASTICSEARCH:
+        final ElasticsearchContainer elasticsearchContainer =
+            TestSearchContainers.createDefeaultElasticsearchContainer()
+                .withNetwork(network)
+                .withNetworkAliases("elasticsearch");
+        elasticsearchContainer.start();
+        dbClosables.add(elasticsearchContainer);
+        databaseExternalUrls.put(db, "http://" + elasticsearchContainer.getHttpHostAddress());
+        break;
+      case OPENSEARCH:
+        final OpensearchContainer opensearchContainer =
+            TestSearchContainers.createDefaultOpensearchContainer()
+                .withNetwork(network)
+                .withNetworkAliases("opensearch");
+        opensearchContainer.start();
+        databaseExternalUrls.put(db, opensearchContainer.getHttpHostAddress());
+        dbClosables.add(opensearchContainer);
+        break;
+      default:
+        throw new IllegalArgumentException("Unsupported database type: " + db);
+    }
+  }
+
+  private void awaitExporterReadiness(
+      final MigrationDatabaseChecks tester, final String indexPrefix) {
+    Awaitility.await("Await database and exporter readiness")
+        .timeout(Duration.of(1, ChronoUnit.MINUTES))
+        .pollInterval(Duration.of(1, ChronoUnit.MILLIS))
+        .untilAsserted(() -> tester.validateSchemaCreation(indexPrefix));
+  }
+
+  private void awaitImportersFinished(
+      final MigrationDatabaseChecks tester, final String indexPrefix) {
+    Awaitility.await("Await Importers finished")
+        .timeout(Duration.of(1, ChronoUnit.MINUTES))
```

Comment:
:upside_down_face: 
```suggestion
        .timeout(Duration.ofMinutes(1))
```

Goes for others `Duration.of`, there's `Duration.ofSeconds`, etc.